spyder/plugins/variableexplorer/widgets/collectionseditor.py

```diff
@@ -1435,13 +1434,9 @@ def get_value(self, name):
     def new_value(self, name, value):
         """Create new value in data"""
         try:
-            # We need to enclose values in a list to be able to send
-            # them to the kernel in Python 2
-            svalue = [cloudpickle.dumps(value, protocol=PICKLE_PROTOCOL)]
-
             # Needed to prevent memory leaks. See spyder-ide/spyder#7158.
-            if len(svalue) < MAX_SERIALIZED_LENGHT:
-                self.shellwidget.set_value(name, svalue)
+            if sys.getsizeof(value) < MAX_SERIALIZED_LENGHT:

```

Comment:
Are you sure that `getsizeof` is equivalent to the size of the serialized object? It's very important to have this right to avoid the serious memory leaks we had in the past.

Reply:
So what are these memory leaks caused by? Is `MAX_SERIALIZED_LENGHT` an estimation of the maximum size or a hard limit?
`sys.getsizeof(value)` is probably similar to the size of the pickled object.
If anything that transit should be below this limit, I should add that to the comms.

Reply:
```python
a = [0] * 10000
len(cloudpickle.dumps(a)) = 20034
sys.getsizeof(a) = 80064
a = np.zeros(10000)
len(cloudpickle.dumps(a)) = 80162
sys.getsizeof(a) = 80096
```

Reply:
On a second thought, this probably doesn't apply to comms but only to our previous architecture. We needed this check because we were executing commands in the kernel with strings, and we noticed that trying to do that for large strings (in this case, large serialized values), was freezing Spyder.

However, I think this should stay on this side and not on the kernel one because we should still prevent users trying to change values on big objects (e.g. big dataframes). Setting values consumes a lot of memory because values have to be serialized here and deserialized in the kernel.

So, could you revert your last commit here and also the one you added in spyder-ide/spyder-kernels#111? Thanks!

Reply:
And giving your comparisons, I think using `getsizeof` is fine.

Reply:
Well actually I realized:
``` python
a = [[0] * 10000]
len(cloudpickle.dumps(a)) = 20034
sys.getsizeof(a) = 72
```
But I can also do a recursive check.

Reply:
And am I missing something or was `len(svalue)` always 1? svalue is an array with one element, that is the buffer. So was this condition ever triggered?

Reply:
Can I add a new dependency? https://pympler.readthedocs.io/en/latest/asizeof.html#asizeof

Reply:
```python
from pympler.asizeof import asizeof
import sys
import cloudpickle
a = [[0] * 1000]
asizeof(a)
8160
sys.getsizeof(a)
72
len(cloudpickle.dumps(a))
2019
```

Reply:
```python
a = [[0] * 1000000]
%timeit len(cloudpickle.dumps(a))
1.65 s
%timeit asizeof(a)
728 ms
```

Reply:
> And am I missing something or was len(svalue) always 1?

Right, but that was a mistake that went unnoticed. We should have checked for `svalue[0]` instead.

> Can I add a new dependency?

Yes, but you need to add it to conda-forge first. [Here](https://github.com/conda-forge/staged-recipes) you can find the instructions on how to do that.

Reply:
Alternatively I can just use `len(cloudpickle.dumps(a))`

Reply:
I don't like that because it we would be serializing `value` twice (first to check its size and then to send it to the kernel). Besides, that check is very slow compared to `asizeof` (as you showed above).

Before it was fine for us to do that because we had to serialize `value` at this point before sending it to the kernel. But now it doesn't make sense to me.

Reply:
ok I'll figure out how to send pympler to  conda-forge

Reply:
It is already there https://anaconda.org/conda-forge/pympler